develop/how-to-contribute/index.md

```diff
@@ -0,0 +1,50 @@
+
+## Discuss the Changes You Want to Do
+
+So once you've decided what you want to do, the very first thing to do is to post to [wx-dev mailing list](https://www.wxwidgets.org/support/mailing-lists/) a message describing what you would like to do and why. This is really important as it allows you to coordinate the activity of different developers and can also allow you to discover better ways of doing what you need or maybe even realize that it has already been done in the latest wxWidgets version--nothing more wasteful than to work on some feature which has already been implemented by someone else. It would also be nice if you could introduce yourself, at least briefly, in your first posting, so that the others could know you better.
+
+Please note that it's a very good idea to subscribe to this mailing list if you plan to become a regular wxWidgets contributor. It is relatively low traffic and you can also browse the archives on the [Google Groups page](https://groups.google.com/forum/#!forum/wx-dev).
+
+Finally, please respect the standard [netiquette guidelines](https://en.wikipedia.org/wiki/Netiquette) for your posts to wx-dev mailing list. In particular, please don't post in (pure) HTML and don't top post.
+
+## Implement the Changes
+
+Once the agreement is reached on the mailing list about the best way to implement the changes, it's time to... not to start coding, but to read  [wxWidgets coding standards](/develop/coding-guidelines/) guide.  Please do it _before_ starting to write the code meant for inclusion in wxWidgets.
+
+### Writing the Documentation
+
+If you submit a new feature (as opposed to just a bug fix), you **must** provide documentation for it. It is better to document the code first and then write the implementation according to the specification, which is why this step comes before writing the code itself. Take a look over the [guide to adding wxWidgets class documentation](https://github.com/wxWidgets/wxWidgets/blob/master/docs/contributing/how-to-add-class-documentation.md).
+
+### Writing the Code
+
+You should try to compile your code under as many platforms as possible (unless it's a port-specific change, of course).  Using different compilers on different architectures ([at maximal warning level](/develop/coding-guidelines/#no_warnings) of course) is a great way to find bugs. Also, implementing a new feature for more than one platform is highly desirable as a single-platform implementation could result in an API impossible or very difficult to implement elsewhere. Changes implemented and tested on multiple platforms have a much higher chance of being accepted quickly in wxWidgets.
+
+If you have any questions during this step, such as "How can I add a new file to wxWidgets?" or "Is there already a generic way to do this and that?" please don't hesitate to ask on wx-dev.
+
+### Writing the Test or a Sample
```

Comment:
Maybe add a reference to 
`[How to write unit tests for wxWidgets][(https://github.com/wxWidgets/wxWidgets/blob/master/docs/contributing/how-to-add-class-documentation.md](https://github.com/wxWidgets/wxWidgets/blob/master/docs/contributing/how-to-write-unit-tests.md))`

In fact, it may not be a bad idea to suggestreading relevant documents in docs/contributing in section Implement the Changes

Reply:
I think this text could be significantly reworked with more links and what not, but I'd like to commit this now in order to able to link to it from other places, so I'm not going to change it now. I will try to do it later (after 3.1.6 but before 3.2.0), but any PRs are, of course, welcome as well.